app/src/main/java/com/example/mymovie/MovieFragment.java

```diff
@@ -0,0 +1,90 @@
+package com.example.mymovie;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+
+public class MovieFragment extends Fragment {
+
+    private ImageView image;
+    private TextView name, detail;
+    private Button detailBtn;
+    private int resId;
+    private String movieName, rate;
+
+    public final static String MOVIE = "movie";
+
+
+    public interface detailCallback{
+        public abstract void onDetail(int resId, String name, String rate);
+    }
+
+    public detailCallback callback;
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if(getParentFragment() instanceof detailCallback){
+            callback = (detailCallback)getParentFragment();
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        callback = null;
+    }
+
+    public static final MovieFragment newInstance(int image, String name, String rate){
+        MovieFragment fragment = new MovieFragment();
+        Bundle bundle = new Bundle(1);
+        bundle.putParcelable(MOVIE, new MovieData(image, name, rate));
+        fragment.setArguments(bundle);
+        return fragment;
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_movie, container,false);
+
+        image = rootView.findViewById(R.id.movie_image);
+        name = rootView.findViewById(R.id.movie_name);
+        detail = rootView.findViewById(R.id.movie_detail);
+        detailBtn = rootView.findViewById(R.id.movie_detail_btn);
```

Comment:
변수명을 초기화 하는 부분입니다. 아래 setData() 처럼 하나의 함수 안에 처리하는건 어떨까요?
```suggestion
        initView() or setupView()
```